core/profile.c

```diff
@@ -1544,10 +1543,15 @@ void compare_samples(const struct dive *d, const struct plot_info *pi, int idx1,
 	avg_depth = 0;
 	max_depth = 0;
 	min_depth = INT_MAX;
-	bar_used = 0;
 
 	last_sec = start->sec;
-	last_pressure = get_plot_pressure(pi, idx1, 0);
+
+	volume_t cylinder_volume = { .mliter = 0, };
+	int *start_pressures = calloc((size_t)pi->nr_cylinders, sizeof(int));
```

Comment:
Not in this PR, but IMHO we really should switch that thing to C++ and start using `std::vector` erc.